### PR TITLE
BE: User can be created and listed

### PR DESCRIPTION
**Pull request by shivamraspro**
The User model is updated and get and post request are working for User entity.

### CODE REVIEW

**Review by vishivish18 — CHANGES_REQUESTED** [review]
Cool
Things remaining:

- UI for the login
- UI for the registration
- API and UI for creating team
- API and UI for adding user to team
- Handle e-mail invitattion

**Comment by vishivish18** [comment]
@shivamras304  Also you have not handled the authentication yet, right ?

**Comment by vishivish18** [comment]
@shivamras304 Whatever commit you make, the comment has to refer to a Issue (if one exists for it) so that you can close this with comment and merge.

**Comment by shivamraspro** [comment]
> @shivamras304 Whatever commit you make, the comment has to refer to a Issue (if one exists for it) so that you can close this with comment and merge.

Do you mean that each commit should correspond to a task for the week so that that task can be closed with the commit?

**Comment by shivamraspro** [comment]
> @shivamras304 Also you have not handled the authentication yet, right ?

No, not yet.

**Comment by shivamraspro** [comment]
Hi Vishal
I have added API for User Registration and User Login.
Please check and merge the code.

File: quiz-app-api/app.js [hunk 1]
```diff
@@ -5,14 +5,15 @@
 "use-strict";
 
 const express = require("express");
-const mongoose = require("mongoose");
+// const mongoose = require("mongoose");
```
Comment (vishivish18) [inline quiz-app-api/app.js:8]: If it's not being used in this file remove it.

File: quiz-app-api/app.js [hunk 2]
```diff
@@ -5,14 +5,15 @@
...
 const routes = require("./app/routes");
+const morgan = require("morgan")
 
 const app = express();
-const router = require("express").Router();
-const path = require("path");
-
+// const router = require("express").Router();
+// const path = require("path");
+app.use(morgan("dev"))
```
Comment (vishivish18) [inline quiz-app-api/app.js:16]: Good choice for logging 👍

File: quiz-app-api/app/controllers/api/UserController.js [hunk 3]
```diff
@@ -1,107 +1,130 @@
...
-        if (err) {
-          console.error(err);
+  signup(req, res) {
+    // If the email already exists, don't signup a new email
+    User.findOne({ email: req.body.data.email })
+      .exec()
+      .then(result => {
+        if (result) {
+          return res.status(409).json({
+            message: 'Email Already Exists',
```
Comment (vishivish18) [inline quiz-app-api/app/controllers/api/UserController.js:21]: Good for now, but later let's create a method which returns appropriate "message text" based on status code. So if someone wants to change message only one place change is required

**Pull request merged**